### PR TITLE
JPERF-273: Generalize Jira installation API

### PR DESCRIPTION
**Pull request by dagguh**

### CODE REVIEW

File: src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/install/InstalledJira.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import com.atlassian.performance.tools.infrastructure.api.jvm.JavaDevelopmentKit
+
+/**
+ * Points to an already installed Jira.
+ *
+ * @since 4.18.0
+ */
+class InstalledJira(
+    /**
+     * Contains `./dbconfig.xml` and other server-specific files.
```
Comment (mzyromski-atlassian) [inline src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/install/InstalledJira.kt:12]: It's not clear, at first glance, that the attribute stores a path (as its type is plain String). Maybe we could note that either in the ktdoc or reflect it in the property name?

**Review by mzyromski-atlassian — APPROVED** [review]
Great work!

File: src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/install/InstalledJira.kt (hunk 1 above)
Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/install/InstalledJira.kt:13]: Good point! I enriched the type.

**Comment by dagguh** [comment]
Resolved conflicts

File: src/test/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/install/ParallelInstallationIT.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        testOnServer { server ->
+            // when
+            val installed = installation.install(server)
+
+            // then
+            val serverXml = installed
+                .installation
+                .resolve("conf/server.xml")
+                .download(Files.createTempFile("downloaded-config", ".xml"))
```
Comment (dagguh) [inline src/test/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/install/ParallelInstallationIT.kt:32]: That's what we get when we upgrade a primitive `String` to an object 😃 
Good catch about ambiguity of `installation: String` @mzyromski-atlassian 👍

**Review by mzyromski-atlassian — APPROVED** [review]

**Pull request merged**